src/common/WebLink.js

```diff
@@ -35,11 +37,13 @@ const componentStyles = createStyleSheet({
 export default function WebLink(props: Props): React.Node {
   const { children } = props;
 
+  const globalSettings = useGlobalSelector(getGlobalSettings);
+
   return (
     <ZulipText
       style={componentStyles.link}
       onPress={() => {
-        openLinkEmbedded(props.url.toString());
+        openLinkWithUserPreference(props.url.toString(), globalSettings);
```

Comment:
I took a look at the places we use this `WebLink` component, and this change seems good for all of them:
```
$ gr -C4 '<WebLink'
src/account-info/CustomProfileFields.js
69-    case 'link':
70-      valueElement = (
71-        <View style={styles.valueView}>
72-          {value.url ? (
73:            <WebLink url={value.url}>
74-              <ZulipText text={value.text} />
75-            </WebLink>
76-          ) : (
77-            <ZulipText text={value.text} />

src/message/ReadReceiptsScreen.js
137-}`,
138-            values: {
139-              num_of_people: displayUserIds.length,
140-              'z-link': chunks => (
141:                <WebLink url={new URL('/help/read-receipts', auth.realm)}>
142-                  {chunks.map(chunk => (
143-                    <ZulipText>{chunk}</ZulipText>
144-                  ))}
145-                </WebLink>

src/start/PasswordAuthScreen.js
142-        />
143-        <ErrorMsg error={error} />
144-        <View style={styles.linksTouchable}>
145-          <ZulipText style={styles.forgotPasswordText}>
146:            <WebLink url={new URL('/accounts/password/reset/', realm)}>
147-              <ZulipTextIntl text="Forgot password?" />
148-            </WebLink>
149-          </ZulipText>
150-        </View>
```